### ISSUE TITLE
Move `sliding_window_view` to the main module and add `step_size`

### ISSUE THREAD

**Issue by tomergt45** [numpy/numpy#18244]
1. Writing `np.lib.stride_tricks.sliding_window_view` is a bit of a mouthful, I think it would be better if it was placed inside the main module. (also doing `import from` is a bit inconvenient)
2. I suggest adding a `step_size` parameter to the function, an example of its use would be:

```
x = np.arange(10)
sliding_window_view(x, 3, step_size=2) 

>> array([[0, 1, 2],
          [2, 3, 4],
          [4, 5, 6],
          [6, 7, 8]])
```

**Comment by eric-wieser** [comment]
I argued against step_size when we designed this function, because you can get that behavior 
already with `sliding_window_view(x, 3)[::2]`.

**Comment by bartbroere** [comment]
Yes, I would also like to see `sliding_window_view` more exposed. This probably means putting a stub in ``numpy/lib/__init__.pyi`` and including it in the `__all__` of `numpy/lib/stride_tricks.py`. I'm not sure if any of these files are automatically generated, but I've added them manually in my own numpy. I would be willing to make a PR of this quick fix. (I'm not sure if the more dangerous `as_strided` should be exposed.)

My use case also needs a step size, but as @eric-wieser suggested, I have achieved that with slicing. This shouldn't be more expensive, since the result is a _view_ on the array already.

Shall I make a PR for this?

**Comment by eric-wieser** [comment]
Note that this was deliberately moved out of the main namespace in #17720, in response to a comment by @rgommers that I can't find any more.

**Comment by bartbroere** [comment]
Ah apologies in that case, I didn't know that. No plans to completely remove this feature hopefully 😅 ?

**Comment by rgommers** [comment]
> Yes, I would also like to see `sliding_window_view` more exposed. This probably means putting a stub in `numpy/lib/__init__.pyi` and including it in the `__all__` of `numpy/lib/stride_tricks.py`.

A  reorganization of the `numpy.lib` namespace is in the works to make that namespace more useful and discoverable, see gh-18447 and https://mail.python.org/pipermail/numpy-discussion/2021-February/081527.html. That should help.

> No plans to completely remove this feature hopefully :sweat_smile: ?

No, we just added it in the most recent release.

**Comment by tomergt45** [comment]
Hi @rgommers, I’m curious to know if there has been any development regarding the mentioned reorganization?

I find myself keep going back to how to implement a sliding window properly, and numpy’s seems to mostly satisfy my needs, I say mostly since it’s location is annoying to remember and the function name is too long, if possible I would suggest renaming it to ‘window’.

**Comment by rgommers** [comment]
> Hi @rgommers, I’m curious to know if there has been any development regarding the mentioned reorganization?

gh-18447 made some initial steps, but it seems to have stalled. 

>  if possible I would suggest renaming it to `window`.

That's way too generic a name; even in numpy itself we have other "window functions" like `bartlett`, `hanning`, etc.
